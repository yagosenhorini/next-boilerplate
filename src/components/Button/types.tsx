export type IButton = {
  backgroundColor?: string;
};
