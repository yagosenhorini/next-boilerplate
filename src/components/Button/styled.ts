import styled from 'styled-components';

export const Button = styled.button<{
  $backgroundColor?: string;
  $isPrimary?: boolean;
}>`
  width: 100px;
  height: 50px;
  border: none;
  font-weight: 700;
  color: ${({ theme }) => theme.colors.white};
  background-color: ${({ theme, $backgroundColor }) =>
    !$backgroundColor ? theme.colors.orange : $backgroundColor};
  border-radius: ${({ $isPrimary }) => ($isPrimary ? '20px' : 'unset')};
`;
