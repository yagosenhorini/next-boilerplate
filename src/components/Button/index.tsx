import * as S from './styled';
import { IButton } from './types';

const Button = ({ backgroundColor }: IButton) => (
  <S.Wrapper>
    <button style={{ backgroundColor: backgroundColor }}>Button</button>
  </S.Wrapper>
);

export default Button;
