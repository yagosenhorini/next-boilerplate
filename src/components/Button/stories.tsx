import { Meta, StoryObj } from '@storybook/react';
import Button from '.';

export default {
  title: 'Button',
  component: Button,
  tags: ['autodocs'],
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as Meta;

export const Default: StoryObj = {};
