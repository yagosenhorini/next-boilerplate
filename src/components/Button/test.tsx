import { render, screen } from '@testing-library/react';

import Button from '.';
import StyledProvider from '../../lib/theme';

describe('<Button />', () => {
  it('should render the button', () => {
    const { container } = render(
      <StyledProvider>
        <Button />
      </StyledProvider>
    );

    expect(screen.getByRole('button', { name: /Button/i })).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
