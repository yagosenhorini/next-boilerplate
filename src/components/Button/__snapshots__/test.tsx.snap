// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the button 1`] = `
<main
  class=""
>
  <button>
    Button
  </button>
</main>
`;
